src/loki_cluster.py

```diff
@@ -0,0 +1,299 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical
+# See LICENSE file for licensing details.
+
+"""This module contains an endpoint wrapper class for the requirer side of the ``loki-cluster`` relation.
+
+As this relation is cluster-internal and not intended for third-party charms to interact with `loki-coordinator-k8s`, its only user will be the loki-worker-k8s charm. As such, it does not live in a charm lib as most other relation endpoint wrappers do.
+"""
```

Comment:
In cos-proxy we have the same situation with [vector](https://github.com/canonical/cos-proxy-operator/blob/8174714a6ac85b95b8db2c3ab0c04ae5b4425b14/lib/charms/vector/v0/vector.py#L26) and [nrpe-exporter](https://github.com/canonical/cos-proxy-operator/blob/8174714a6ac85b95b8db2c3ab0c04ae5b4425b14/lib/charms/nrpe_exporter/v0/nrpe_exporter.py#L46) and we still put them under lib.
A bit confusing but tidy? Wdyt?

Reply:
I agree, it can even be a real lib.

Following here:

https://github.com/canonical/loki-worker-k8s-operator/issues/5